src/client/interpreter/locators/services/condaEnvFileService.ts

```diff
@@ -0,0 +1,69 @@
+"use strict";
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { IInterpreterVersionService } from '../../interpreterVersion';
+import { IInterpreterLocatorService, PythonInterpreter } from '../../contracts';
+import { AnacondaDisplayName, AnacondaCompanyName, CONDA_RELATIVE_PY_PATH } from './conda';
+
+/**
+ * The command 'conda info --envs' doesn't seem to return all environments
+ * Environments created using the command 'conda create --p python=x.x' are not returned by the above command
+ * However all these environments seem to be listed in the environments.txt file (confirmed on windows and linux)
+ * @export
+ * @class CondaEnvFileProvider
+ * @implements {IInterpreterLocatorService}
+ */
+export class CondaEnvFileService implements IInterpreterLocatorService {
+    constructor(private condaEnvironmentFile: string,
+        private versionService: IInterpreterVersionService) {
+    }
+    public getInterpreters() {
+        return this.getSuggestionsFromConda();
+    }
+
+    private getSuggestionsFromConda(): Promise<PythonInterpreter[]> {
+        return fs.pathExists(this.condaEnvironmentFile)
+            .then(exists => exists ? this.getEnvironmentsFromFile(this.condaEnvironmentFile) : Promise.resolve([]));
+    }
+    private getEnvironmentsFromFile(envFile: string) {
+        return fs.readFile(envFile)
+            .then(buffer => buffer.toString().split(/\r?\n/g))
+            .then(lines => {
+                return lines.map(line => path.join(line, ...CONDA_RELATIVE_PY_PATH));
+            })
+            .then(interpreterPaths => {
+                return interpreterPaths.map(item => fs.pathExists(item).then(exists => exists ? item : ''));
+            })
+            .then(promises => Promise.all(promises))
+            .then(interpreterPaths => {
+                return interpreterPaths.filter(item => item.trim().length > 0);
```

Comment:
Is it worth keeping the paths trimmed instead of tossing the trimmed result? I.e.
```typescript
interpreterPaths.map(String.prototype.trim).filter(item => item.length > 0);
```

Reply:
Agreed